tensorflow/java/src/main/java/org/tensorflow/op/OperationHelper.java

```diff
@@ -0,0 +1,147 @@
+/* Copyright 2017 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+package org.tensorflow.op;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import org.tensorflow.Operation;
+import org.tensorflow.OperationBuilder;
+import org.tensorflow.Output;
+
+/**
+ * A helper for wrapping a new operation.
+ *
+ * <p>It provides utilities for wrappers to build a single operation and collect its output(s)
+ * afterward.
+ *
+ * <pre>{@code
+ * public final class Unique extends AbstractSingleOp {
+ *    ...
+ *    public static Unique create(Scope s, Input x) {
+ *        OperationHelper unique = OperationHelper.create(s, "Unique");
+ *        unique.builder().input(x.asOutput());
+ *        return new Unique(unique);
+ *    }
+ *    ...
+ *    private Unique(OperationHelper unique) {
+ *        super(unique.operation());
+ *        y = unique.nextOutput();
```

Comment:
Perhaps flesh this example out in more detail (what are `y` and `idx` for example).